tools/build-shader-cache.py

```
@@ -204,6 +204,29 @@ def check_default_disabled(source_root: Path, feature_versions_header: Path) ->
     return 1
 
 
+def check_default_disabled_stage_list(source_root: Path, stage_list_file: Path) -> int:
+    """Cheaper sibling of check_default_disabled(): compares against a plain
+    feature-name list (one per line, from cmake/emit-default-disabled-features.cmake)
+    instead of a configured build's generated FeatureVersions.h. Same drift this
+    guards against (see check_default_disabled's docstring), but the CMake side
+    reuses cmake/FeatureStaging.cmake standalone (no project()/vcpkg configure), so
+    this can run in the same cheap, cache-free CI job as --check-aio-partition."""
+    cmake_set = {l.strip() for l in stage_list_file.read_text(encoding="utf-8").splitlines() if l.strip()}
```

Ruff (0.16.1)

[error] Ambiguous variable name: `l`

(E741)